packages/backend-tasks/src/tasks/PluginTaskSchedulerImpl.ts

```diff
@@ -35,16 +40,25 @@ export class PluginTaskSchedulerImpl implements PluginTaskScheduler {
     const knex = await this.databaseFactory();
 
     const worker = new TaskWorker(task.id, task.fn, knex, this.logger);
-    await worker.start(
-      {
-        version: 1,
-        initialDelayDuration: task.initialDelay?.toISO(),
-        recurringAtMostEveryDuration: task.frequency.toISO(),
-        timeoutAfterDuration: task.timeout.toISO(),
-      },
-      {
-        signal: task.signal,
-      },
-    );
+
+    const settings: any = {
```

Comment:
Let's not use `any` (basically, almost never :))

You could make two `const`s out of these values if you want to reuse them, and maybe a `const settings: TaskSettingsV1 | TaskSettingsV2;` here, but assign `settings = { ... full contents }` with its entire body at once inside each `if` option below so you can see all of it happening in one place, collected.

Reply:
Ah fair enough!

Reply:
Thanks for the catch, this should be a non-issue with the latest changes to move completely to TaskSettingsV2